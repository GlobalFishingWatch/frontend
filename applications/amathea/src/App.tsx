import React, { memo, useState, Fragment } from 'react'
import { useSelector } from 'react-redux'
import SplitView from '@globalfishingwatch/ui-components/dist/split-view'
import Login from './features/user/Login'
import Map from './features/map/Map'
import Timebar from './features/timebar/Timebar'
import Sidebar from './features/sidebar/Sidebar'
import styles from './App.module.css'
import { isUserLogged } from './features/user/user.slice'

import '@globalfishingwatch/ui-components/dist/base.css'

const Main = memo(() => (
  <div className={styles.main}>
    <Map />
    <Timebar />
  </div>
))

function App(): React.ReactElement {
  const [sidebarOpen, setSidebarOpen] = useState(true)
  const logged = useSelector(isUserLogged)

  const onToggle = () => {
    setSidebarOpen(!sidebarOpen)
  }

  return (
    <Fragment>
      <Login />
      {!logged ? (
        <div className={styles.placeholder}>Loading</div>
      ) : (
        <SplitView
          isOpen={sidebarOpen}
          onToggle={onToggle}
          aside={<Sidebar />}
          main={<Main />}
          asideWidth="30%"
          className="split-container"
        />
      )}
    </Fragment>
  )
}

export default App
