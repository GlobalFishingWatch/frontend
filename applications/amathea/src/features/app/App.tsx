import React, { memo, Fragment, useCallback, useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import SplitView from '@globalfishingwatch/ui-components/dist/split-view'
import Menu from '@globalfishingwatch/ui-components/dist/menu'
import { useLocationConnect } from 'routes/routes.hook'
import { WORKSPACE_EDITOR } from 'routes/routes'
import { useAOIConnect } from 'features/areas-of-interest/areas-of-interest.hook'
import { useWorkspacesConnect } from 'features/workspaces/workspaces.hook'
import Login from '../user/Login'
import Modal from '../modal/Modal'
import Map from '../map/Map'
import Timebar from '../timebar/Timebar'
import Sidebar from '../sidebar/Sidebar'
import { isUserLogged } from '../user/user.slice'
import { toggleMenu, isMenuOpen, isSidebarOpen, toggleSidebar } from './app.slice'
import styles from './App.module.css'
import '@globalfishingwatch/ui-components/dist/base.css'

const Main = memo(() => {
  const { location } = useLocationConnect()
  return (
    <div className={styles.main}>
      <Map />
      {location.type === WORKSPACE_EDITOR && <Timebar />}
    </div>
  )
})

function App(): React.ReactElement {
  const menuOpen = useSelector(isMenuOpen)
  const sidebarOpen = useSelector(isSidebarOpen)
  const logged = useSelector(isUserLogged)
  const dispatch = useDispatch()

  const { fetchAOI } = useAOIConnect()
  const { fetchWorkspaces } = useWorkspacesConnect()
  useEffect(() => {
    fetchAOI()
    fetchWorkspaces()
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  const onToggleMenu = useCallback(() => {
    dispatch(toggleMenu())
  }, [dispatch])

  const onToggleSidebar = useCallback(() => {
    dispatch(toggleSidebar())
  }, [dispatch])

  return (
    <Fragment>
      <Login />
      {!logged ? (
        <div className={styles.placeholder}>Loading</div>
      ) : (
        <SplitView
          isOpen={sidebarOpen}
          onToggle={onToggleSidebar}
          aside={<Sidebar />}
          main={<Main />}
          asideWidth="50%"
          className="split-container"
        />
      )}
      <Modal />
      <Menu isOpen={menuOpen} onClose={onToggleMenu} activeLinkId="map-data">
        Menu toggle
      </Menu>
    </Fragment>
  )
}

export default App
