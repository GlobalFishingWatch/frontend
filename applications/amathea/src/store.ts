import {
  configureStore,
  ThunkAction,
  Action,
  combineReducers,
  getDefaultMiddleware,
} from '@reduxjs/toolkit'
import datasetsReducer from './features/datasets/datasets.slice'
import appReducer from './features/app/app.slice'
import aoiReducer from './features/areas-of-interest/areas-of-interest.slice'
import workspacesReducer from './features/workspaces/workspaces.slice'
import userReducer from './features/user/user.slice'
import mapReducer from './features/map/map.slice'
import timebarReducer from './features/timebar/timebar.slice'
import connectedRoutes, { routerQueryMiddleware } from './routes/routes'

const {
  reducer: location,
  middleware: routerMiddleware,
  enhancer: routerEnhancer,
  // initialDispatch,
} = connectedRoutes

const rootReducer = combineReducers({
  app: appReducer,
  user: userReducer,
  aoi: aoiReducer,
  datasets: datasetsReducer,
  workspaces: workspacesReducer,
  map: mapReducer,
  timebar: timebarReducer,
  location: location,
})

// Can't type because GetDefaultMiddlewareOptions type is not exposed by RTK
const defaultMiddlewareOptions: any = {
  // Fix issue with Redux-first-router and RTK (https://stackoverflow.com/questions/59773345/react-toolkit-and-redux-first-router)
  serializableCheck: false,
  immutableCheck: {},
}

const store = configureStore({
  reducer: rootReducer,
  middleware: [
    ...getDefaultMiddleware(defaultMiddlewareOptions),
    routerQueryMiddleware,
    routerMiddleware,
  ],
  enhancers: (defaultEnhancers) => [routerEnhancer, ...defaultEnhancers],
})

export type RootState = ReturnType<typeof store.getState>
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>

export default store
