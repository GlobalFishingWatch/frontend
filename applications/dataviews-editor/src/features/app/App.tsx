import React, { useState, Fragment } from 'react'
import cx from 'classnames'
import styles from './App.module.css'
import Dataviews from 'features/dataviews/Dataviews'
import Dataview from 'features/dataview/Dataview'
import Map from 'features/map/Map'
import Timebar from 'features/timebar/Timebar'

const App = () => {
  const [dataviewsMinimized, toggleDataviews] = useState(false)
  const [dataviewMinimized, toggleDataview] = useState(false)
  // console.log(dataviews, 'lol')
  return (
  <div className={styles.app}>
    <div className={cx(styles.column, styles.dataviews, { [styles.minimized]: dataviewsMinimized })}>
      <h1 onClick={() => { toggleDataviews(!dataviewsMinimized) }}>Dataviews</h1>
      {!dataviewsMinimized && (
        <Fragment>
          <Dataviews />
        </Fragment>
      )}
    </div>
    <div className={cx(styles.column, styles.dataview, { [styles.minimized]: dataviewMinimized })}>
      <h1 onClick={() => { toggleDataview(!dataviewMinimized) }}>Dataview</h1>
      {!dataviewMinimized && (
        <Fragment>
          <Dataview />
        </Fragment>
      )}
    </div>
    <div className={styles.map}>
      <Map />
    </div>
    <div className={styles.timebar}>
      <Timebar />
    </div>
  </div>
  )
}

export default App