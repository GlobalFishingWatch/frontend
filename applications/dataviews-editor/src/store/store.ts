import { combineReducers } from 'redux'
import { configureStore } from '@reduxjs/toolkit'
import dataviews from 'features/dataviews/dataviews.slice'
import timebar from 'features/timebar/timebar.slice'

const rootReducer = combineReducers({
  dataviews,
  timebar,
})

export type RootState = ReturnType<typeof rootReducer>

export default () => {
  const store = configureStore({
    reducer: rootReducer,
  })
  // initialDispatch()
  return store
}
