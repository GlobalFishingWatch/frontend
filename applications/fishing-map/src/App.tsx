import React, { memo, useState, Fragment } from 'react'
import { useSelector } from 'react-redux'
import SplitView from '@globalfishingwatch/ui-components/dist/split-view'
import Spinner from '@globalfishingwatch/ui-components/dist/spinner'
import Login from './features/user/Login'
import Map from './features/map/Map'
import Timebar from './features/timebar/Timebar'
import Sidebar from './features/sidebar/Sidebar'
import styles from './App.module.css'
import { isUserLogged } from './features/user/user.slice'

import '@globalfishingwatch/ui-components/dist/base.css'

const Main = memo(() => (
  <div className={styles.main}>
    <Map />
    <Timebar />
  </div>
))

function App(): React.ReactElement {
  const [sidebarOpen, setSidebarOpen] = useState(true)
  const logged = useSelector(isUserLogged)

  const onToggle = () => {
    setSidebarOpen(!sidebarOpen)
  }

  return (
    <Fragment>
      <Login />
      {!logged ? (
        <div className={styles.placeholder}>
          <Spinner />
        </div>
      ) : (
        <SplitView
          isOpen={sidebarOpen}
          onToggle={onToggle}
          aside={<Sidebar />}
          main={<Main />}
          asideWidth="30%"
          className="split-container"
        />
      )}
    </Fragment>
  )
}

export default App
