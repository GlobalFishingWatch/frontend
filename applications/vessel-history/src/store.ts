import { configureStore, ThunkAction, Action, combineReducers } from '@reduxjs/toolkit'
import connectedRoutes, { routerQueryMiddleware } from 'routes/routes'
import offlineVesselsReducer from 'features/vessels/offline-vessels.slice'
import vesselsReducer from 'features/vessels/vessels.slice'
import searchReducer from 'features/search/search.slice'
import settingsReducer from 'features/settings/settings.slice'
import filtersReducer from 'features/profile/filters/filters.slice'
import { initializeDataviews } from 'features/dataviews/dataviews.utils'
import mapReducer from './features/map/map.slice'
import dataviewsReducer from './features/dataviews/dataviews.slice'
import datasetsReducer from './features/datasets/datasets.slice'
import psmaReducer from './features/psma/psma.slice'
import regionsReducer from './features/regions/regions.slice'
import resourcesReducer from './features/resources/resources.slice'
import workspaceReducer from './features/workspace/workspace.slice'

const {
  reducer: location,
  middleware: routerMiddleware,
  enhancer: routerEnhancer,
} = connectedRoutes

const rootReducer = combineReducers({
  offlineVessels: offlineVesselsReducer,
  vessels: vesselsReducer,
  search: searchReducer,
  settings: settingsReducer,
  filters: filtersReducer,
  location: location,
  map: mapReducer,
  dataviews: dataviewsReducer,
  datasets: datasetsReducer,
  psma: psmaReducer,
  regions: regionsReducer,
  resources: resourcesReducer,
  workspace: workspaceReducer,
})

// Can't type because GetDefaultMiddlewareOptions type is not exposed by RTK
const defaultMiddlewareOptions: any = {
  // Fix issue with Redux-first-router and RTK (https://stackoverflow.com/questions/59773345/react-toolkit-and-redux-first-router)
  serializableCheck: false,
  immutableCheck: {
    ignoredPaths: [
      // Too big to check for immutability:
      'resources',
    ],
  },
}

const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware(defaultMiddlewareOptions)
      .concat(routerQueryMiddleware)
      .concat(routerMiddleware),
  enhancers: (defaultEnhancers) => [routerEnhancer, ...defaultEnhancers],
})

export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = typeof store.dispatch
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>

// Only once when the app starts
initializeDataviews(store.dispatch)

export default store
