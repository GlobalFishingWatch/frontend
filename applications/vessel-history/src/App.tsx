import React, { Fragment, useEffect, useState } from 'react'
import { useSelector } from 'react-redux'
import Home from 'features/home/Home'
import { getLocationType } from 'routes/routes.selectors'
import { PROFILE, SETTINGS } from 'routes/routes'
import { SPLASH_TIMEOUT } from 'data/constants'
import Profile from 'features/profile/Profile'
import Splash from 'features/splash/Splash'
import Settings from 'features/settings/Settings'
import { useUser } from 'features/user/user.hooks'
import { useAnalytics } from 'features/app/analytics.hooks'
import './App.css'

function App() {
  useAnalytics()
  const { loading, logged, authorized } = useUser()
  const [minLoading, setMinLoading] = useState(true)

  // Splash screen is shown at least one second
  useEffect(() => {
    const timer = setTimeout(() => setMinLoading(false), SPLASH_TIMEOUT)
    return () => clearTimeout(timer)
  }, [])

  const locationType = useSelector(getLocationType)

  if (loading || minLoading || !logged || !authorized) {
    return <Splash intro={minLoading} />
  }
  if (locationType === SETTINGS) {
    return <Settings />
  }
  if (locationType === PROFILE) {
    return <Profile />
  }
  return (
    <Fragment>
      <Home />
    </Fragment>
  )
}

export default App
