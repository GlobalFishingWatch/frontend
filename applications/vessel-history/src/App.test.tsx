import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'
import useGFWLogin from '@globalfishingwatch/react-hooks/dist/use-login'
import store from './store'
import App from './App'

jest.mock('@globalfishingwatch/react-hooks/dist/use-login')

beforeEach(() => {
  useGFWLogin.mockClear()
})

const assignMock = jest.fn()
delete window.location
window.location = { assign: assignMock }

afterEach(() => {
  assignMock.mockClear()
})

it('renders splash screen while loading', () => {
  useGFWLogin.mockReturnValue({
    loading: true,
    logged: false,
    user: null,
    error: null,
  })
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('redirects to login screen when not logged', () => {
  useGFWLogin.mockReturnValue({
    loading: false,
    logged: false,
    user: null,
    error: null,
  })
  const tree = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  )
  expect(assignMock).toHaveBeenCalled()
})
it('renders home screen when not loading', () => {
  useGFWLogin.mockReturnValue({
    loading: false,
    logged: true,
    user: null,
    error: null,
  })
  const tree = renderer
    .create(
      <Provider store={store}>
        <App />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
