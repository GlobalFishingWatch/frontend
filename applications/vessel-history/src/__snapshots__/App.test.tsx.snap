// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home screen when not loading 1`] = `
<div
  className="homeContainer"
>
  <header>
    <svg
      aria-label="Global Fishing Watch "
      className="Logo-module_container__3qEvx"
      height="60"
      viewBox="40 0 160 60"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        aria-hidden="true"
        fill="none"
        fillRule="evenodd"
        focusable="false"
      >
        <path
          d="M23.1 27.83c-.08.16-2.25 4.5.57 8.76.83 1.25 1.76 2.24 4.1 3.8 3.38 2.26 7.42 6.16 8.4 8.45.1.21.36.59-.06.83-.73.41-1.75.31-2.16.25l-.12-.02c-4.75-.88-14.5-5-13.8-13.8.18-2.21.98-4.87 3.07-8.27zm9.92-8.64c.7-.08 2.98-.15 3.08.08.07.17-.83.5-.87.52-5.22 1.98-6.97 3.16-7.22 3.3-3.3 2.01-4.71 4.35-4.82 4.57 1.48-3.1 5.2-7.93 9.83-8.47z"
          fill="#8ABBC7"
        />
        <path
          d="M40.45 19.2l.64 1.16a17.1 17.1 0 011.52 3.94c.27 1.1.39 2.23.31 3.37a9.39 9.39 0 01-1.72 4.84 12.9 12.9 0 01-3.29 3.2 20.3 20.3 0 01-7.02 3.08c-.4.09-.8.16-1.2.19a3.39 3.39 0 01-1.3-.17c-.43-.17-.5-.43-.28-.83a13.43 13.43 0 012.53-3.4 35.39 35.39 0 015.48-4.56c.76-.53 1.5-1.06 2.15-1.7a8.5 8.5 0 002.55-4.63 8.37 8.37 0 00-.24-3.89l-.18-.59.05-.02zm.64 16.43c.5.03.64.28.4.72-.35.64-.9 1.04-1.6 1.21-.68.16-1.35.13-2.02-.02-.42-.09-.83-.23-1.24-.35l-.11-.06.12-.08c1.02-.53 2.07-.97 3.19-1.26.4-.1.83-.18 1.26-.16zM30.73 10.07c.92.2 1.75.61 2.54 1.11.87.56 1.65 1.21 2.4 1.92a29.78 29.78 0 014.2 5.18l.01.07-.18-.16a14.79 14.79 0 00-3.6-2.45c-1.01-.5-2.06-.9-3.11-1.3-.58-.23-1.16-.44-1.7-.75a1.82 1.82 0 01-.75-.76c-.15-.32-.07-.6.23-.78l.35-.18c-.28-.14-.53-.24-.75-.38-.2-.13-.38-.29-.55-.45a.77.77 0 01-.21-.64c.02-.24.13-.38.35-.46.26-.08.52-.03.77.03z"
          fill="#E84427"
        />
        <text
          className="Logo-module_mainBrand__3OqxN Logo-module_default__2NdMB"
          x="52"
          y="37"
        >
          Global Fishing Watch
        </text>
        
      </g>
    </svg>
    <button
      className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe"
    >
      <svg
        aria-hidden="true"
        className="Icon-module_default__2Pvfr"
        height={20}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.18 2.8h.06l.14.01c.3.07.52.29.63.61l.03.13.2 1.45.24.12.12.07.12.07.23.15 1.44-.57.11-.03c.34-.06.64.05.84.31l.07.1 1.36 2.36.06.11c.13.34.06.7-.19.92l-.09.07-1.17.91v.82l1.2.92.09.09c.25.26.28.6.14.93l-.06.11-1.36 2.35-.08.1c-.21.26-.52.34-.86.26l-.13-.03-1.38-.55-.2.13-.28.17-.13.07-.08.04-.23 1.53-.04.14a.83.83 0 01-.68.53H8.44l-.14-.01c-.3-.07-.52-.29-.63-.61l-.03-.13-.2-1.45-.24-.12-.12-.07-.12-.07-.23-.15-1.44.57-.11.03a.83.83 0 01-.84-.31l-.07-.1-1.36-2.36-.06-.11a.82.82 0 01.19-.92l.09-.07 1.17-.92v-.8l-1.2-.92-.09-.09c-.25-.26-.28-.6-.14-.93l.06-.11L4.29 5.2l.08-.1c.21-.26.52-.34.86-.26l.13.03 1.38.55.2-.13.28-.17.13-.07.08-.04.23-1.53.04-.14c.12-.3.37-.49.68-.53h2.8zm-.2 1.1H8.7l-.27 1.87-.3.12c-.17.07-.38.18-.61.33l-.18.11-.46.32-1.76-.7-1.16 2 1.5 1.16-.04.31c-.02.1-.03.24-.03.43V10l.01.42.02.17.03.31-1.49 1.16 1.16 1.99 1.78-.7.24.18a5 5 0 00.59.39l.2.1.5.23.27 1.86h2.28l.27-1.88.3-.12c.17-.07.38-.18.61-.33l.18-.11.46-.32 1.76.7 1.16-2-1.5-1.16.04-.31c.02-.1.03-.24.03-.43V10l-.01-.42-.02-.17-.03-.31 1.49-1.16-1.16-1.99-1.78.7-.24-.18a5 5 0 00-.59-.39l-.2-.1-.5-.23-.27-1.86zM9.84 7.1c.8 0 1.48.3 2.05.85.56.57.84 1.26.84 2.05 0 .8-.28 1.48-.84 2.05a2.8 2.8 0 01-2.05.84 2.8 2.8 0 01-2.05-.84A2.8 2.8 0 016.95 10c0-.8.28-1.48.84-2.05a2.8 2.8 0 012.05-.84zm0 1.1c-.5 0-.91.18-1.27.53A1.7 1.7 0 008.05 10c0 .5.17.91.52 1.27.36.35.77.52 1.27.52s.91-.17 1.27-.52c.35-.36.52-.77.52-1.27s-.17-.91-.52-1.27a1.7 1.7 0 00-1.27-.52z"
        />
      </svg>
    </button>
    <button
      className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="Icon-module_default__2Pvfr"
        height={20}
        width={20}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.9 15.67c0 .33-.25.6-.56.64l-.1.01h-6a.65.65 0 01-.65-.55V4.8c0-.32.24-.6.55-.64h6.1a.65.65 0 01.1 1.29H4.89v9.56h5.35c.33 0 .6.24.65.56v.1zm-3.63-5.43c0 .33.24.6.56.65h6.36l-1.7 1.71a.65.65 0 00.83.99l.08-.07 2.82-2.82a.65.65 0 00.07-.84l-.07-.08-2.82-2.82a.65.65 0 00-.98.84l.06.08L14.2 9.6H7.92c-.35 0-.65.3-.65.65z"
        />
      </svg>
    </button>
  </header>
  <div>
    <div
      className="searchbar"
    >
      <input
        aria-label="Search vessels"
        autoFocus={true}
        className="input"
        onBlur={[Function]}
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search vessels by name, MMSI, IMO"
        role="search"
        type="search"
        value=""
      />
      <button
        className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe searchButton"
      >
        <svg
          aria-hidden="true"
          className="Icon-module_default__2Pvfr"
          height={20}
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 4.35a4.65 4.65 0 013.715 7.447l3.245 3.243a.65.65 0 01-.84.988l-.08-.068-3.243-3.245A4.65 4.65 0 119 4.35zm0 1.3a3.35 3.35 0 100 6.7 3.35 3.35 0 000-6.7z"
          />
        </svg>
      </button>
    </div>
    <div>
      <h2>
        OFFLINE ACCESS
      </h2>
      <div
        className="offlineVessels"
      >
        <div
          className="vesselItem"
        >
          <button
            className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe deleteSaved"
          >
            <svg
              aria-hidden="true"
              className="Icon-module_default__2Pvfr"
              height={20}
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.739 5.97v9.642c0 .427-.156.804-.468 1.13-.311.327-.67.49-1.078.49H7.08c-.407 0-.767-.163-1.078-.49a1.588 1.588 0 01-.468-1.13V5.97h9.205zm-1.001.999H6.534v8.643c0 .133.035.243.119.355l.072.085c.102.107.188.158.282.174l.073.006h6.113c.125 0 .227-.046.355-.18a.615.615 0 00.184-.344l.007-.096-.001-8.643zm-1.696-4.201l.791.79h2.66v1.62H4.78v-1.62h2.66l.792-.79h3.811z"
              />
            </svg>
          </button>
          <h3>
            Don Ying Gang 8003
          </h3>
          <div
            className="identifiers"
          >
            <div>
              <label>
                FLAG
              </label>
              Chinese Taipei
            </div>
            <div>
              <label>
                MMSI
              </label>
              31306526
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                CALLSIGN
              </label>
              407201
            </div>
            <div>
              <label>
                TRANSMISSIONS
              </label>
              from 2015-07-21 to 2020/04/13
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                SAVED ON
              </label>
              2020/08/01
            </div>
          </div>
        </div>
        <div
          className="vesselItem"
        >
          <button
            className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe deleteSaved"
          >
            <svg
              aria-hidden="true"
              className="Icon-module_default__2Pvfr"
              height={20}
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.739 5.97v9.642c0 .427-.156.804-.468 1.13-.311.327-.67.49-1.078.49H7.08c-.407 0-.767-.163-1.078-.49a1.588 1.588 0 01-.468-1.13V5.97h9.205zm-1.001.999H6.534v8.643c0 .133.035.243.119.355l.072.085c.102.107.188.158.282.174l.073.006h6.113c.125 0 .227-.046.355-.18a.615.615 0 00.184-.344l.007-.096-.001-8.643zm-1.696-4.201l.791.79h2.66v1.62H4.78v-1.62h2.66l.792-.79h3.811z"
              />
            </svg>
          </button>
          <h3>
            Don Ying Gang 8003
          </h3>
          <div
            className="identifiers"
          >
            <div>
              <label>
                FLAG
              </label>
              Chinese Taipei
            </div>
            <div>
              <label>
                MMSI
              </label>
              31306526
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                CALLSIGN
              </label>
              407201
            </div>
            <div>
              <label>
                TRANSMISSIONS
              </label>
              from 2015-07-21 to 2020/04/13
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                SAVED ON
              </label>
              2020/08/01
            </div>
          </div>
        </div>
        <div
          className="vesselItem"
        >
          <button
            className="IconButton-module_iconButton__cB0-J IconButton-module_default__nAYJm IconButton-module_defaultSize__1OZDe deleteSaved"
          >
            <svg
              aria-hidden="true"
              className="Icon-module_default__2Pvfr"
              height={20}
              width={20}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.739 5.97v9.642c0 .427-.156.804-.468 1.13-.311.327-.67.49-1.078.49H7.08c-.407 0-.767-.163-1.078-.49a1.588 1.588 0 01-.468-1.13V5.97h9.205zm-1.001.999H6.534v8.643c0 .133.035.243.119.355l.072.085c.102.107.188.158.282.174l.073.006h6.113c.125 0 .227-.046.355-.18a.615.615 0 00.184-.344l.007-.096-.001-8.643zm-1.696-4.201l.791.79h2.66v1.62H4.78v-1.62h2.66l.792-.79h3.811z"
              />
            </svg>
          </button>
          <h3>
            Don Ying Gang 8003
          </h3>
          <div
            className="identifiers"
          >
            <div>
              <label>
                FLAG
              </label>
              Chinese Taipei
            </div>
            <div>
              <label>
                MMSI
              </label>
              31306526
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                CALLSIGN
              </label>
              407201
            </div>
            <div>
              <label>
                TRANSMISSIONS
              </label>
              from 2015-07-21 to 2020/04/13
            </div>
            <div>
              <label>
                IMO
              </label>
              12345
            </div>
            <div>
              <label>
                SAVED ON
              </label>
              2020/08/01
            </div>
          </div>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;

exports[`renders splash screen while loading 1`] = `
<div
  className="container"
>
  <div
    className="splash"
  >
    <img
      alt="Vessel History"
      className="vhLogo"
      src="splash-screen-image@2x.png"
    />
  </div>
  <div
    className="secondary"
  >
    <img
      alt="Trygg Mat Tracking"
      className="tmtLogo"
      src="tmt_logo_final_full_colour@2x.png"
    />
    <svg
      aria-label="Global Fishing Watch "
      className="Logo-module_container__3qEvx gfwLogo"
      height="60"
      viewBox="40 0 160 60"
      width="200"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        aria-hidden="true"
        fill="none"
        fillRule="evenodd"
        focusable="false"
      >
        <path
          d="M23.1 27.83c-.08.16-2.25 4.5.57 8.76.83 1.25 1.76 2.24 4.1 3.8 3.38 2.26 7.42 6.16 8.4 8.45.1.21.36.59-.06.83-.73.41-1.75.31-2.16.25l-.12-.02c-4.75-.88-14.5-5-13.8-13.8.18-2.21.98-4.87 3.07-8.27zm9.92-8.64c.7-.08 2.98-.15 3.08.08.07.17-.83.5-.87.52-5.22 1.98-6.97 3.16-7.22 3.3-3.3 2.01-4.71 4.35-4.82 4.57 1.48-3.1 5.2-7.93 9.83-8.47z"
          fill="#8ABBC7"
        />
        <path
          d="M40.45 19.2l.64 1.16a17.1 17.1 0 011.52 3.94c.27 1.1.39 2.23.31 3.37a9.39 9.39 0 01-1.72 4.84 12.9 12.9 0 01-3.29 3.2 20.3 20.3 0 01-7.02 3.08c-.4.09-.8.16-1.2.19a3.39 3.39 0 01-1.3-.17c-.43-.17-.5-.43-.28-.83a13.43 13.43 0 012.53-3.4 35.39 35.39 0 015.48-4.56c.76-.53 1.5-1.06 2.15-1.7a8.5 8.5 0 002.55-4.63 8.37 8.37 0 00-.24-3.89l-.18-.59.05-.02zm.64 16.43c.5.03.64.28.4.72-.35.64-.9 1.04-1.6 1.21-.68.16-1.35.13-2.02-.02-.42-.09-.83-.23-1.24-.35l-.11-.06.12-.08c1.02-.53 2.07-.97 3.19-1.26.4-.1.83-.18 1.26-.16zM30.73 10.07c.92.2 1.75.61 2.54 1.11.87.56 1.65 1.21 2.4 1.92a29.78 29.78 0 014.2 5.18l.01.07-.18-.16a14.79 14.79 0 00-3.6-2.45c-1.01-.5-2.06-.9-3.11-1.3-.58-.23-1.16-.44-1.7-.75a1.82 1.82 0 01-.75-.76c-.15-.32-.07-.6.23-.78l.35-.18c-.28-.14-.53-.24-.75-.38-.2-.13-.38-.29-.55-.45a.77.77 0 01-.21-.64c.02-.24.13-.38.35-.46.26-.08.52-.03.77.03z"
          fill="#E84427"
        />
        <text
          className="Logo-module_mainBrand__3OqxN Logo-module_default__2NdMB"
          x="52"
          y="37"
        >
          Global Fishing Watch
        </text>
        
      </g>
    </svg>
  </div>
</div>
`;
